scripts/anvil_tools/split_and_push_data_model_tsvs.py

```diff
@@ -0,0 +1,66 @@
+"""Split single data model load tsv file by workspace_name (and workspace_project) into individual load tsv files and push to respective Terra workspace.
+
+Usage:
+    > python3 split_and_push_data_model_tsvs.py -t TSV_FILE [-a ARRAY_COLUMNS_FILE]"""
+
+import argparse
+import pandas as pd
+import batch_upsert_entities_standard
+
+
+def split_and_push_workspace_entities(tsv, array_column_names=None):
+    """Create individual request body per unique project/workspace noted in tsv file and push request/data table to workspace."""
+
+    # read full tsv into dataframe - contains all rows for all data across all workspaces
+    tsv_all = pd.read_csv(tsv, sep="\t")
+
+    # start the array attribute column list as empty - fill in if user provides a file with list of column names of array type attributes
+    array_attr_cols = []
+    if array_column_names:
+        # read array column names into list - strip out \n
+        with open(array_column_names, "r") as f:
+            array_attr_cols = [line.strip() for line in f]
+
+    # get unique list of tuples where each tuple = (workspace_name, workspace_project)
+    # length of list = number of workspaces for which tsvs will be generated and pushed via FISS
+    all_workspaces = list(tsv_all[['workspace_name', 'workspace_project']].drop_duplicates().to_records(index=False))
+
+    # for each tuple in list (workspace, project)
+    for workspace in all_workspaces:
+        workspace_name = workspace[0]
+        workspace_project = workspace[1]
+
+        # get rows that match the combination of workspace_name and workspace_project, drop the workspace identifying columns (name + project)
+        workspace_tsv = tsv_all.loc[(tsv_all['workspace_name'] == workspace_name) & (tsv_all['workspace_project'] == workspace_project)] \
+            .drop(['workspace_name', 'workspace_project'], axis=1)
+
+        # pass in workspace subsetted tsv to convert the df into properly formatted batchUpsert request
+        print(f"Starting entity updates to {workspace_project}/{workspace_name}:")
+        print(f"Creating json request.")
+        json_request = batch_upsert_entities_standard.create_upsert_request(workspace_tsv, array_attr_cols)
+
+        # write out a json of the request body
```

Comment:
why do you need to write this local file? is this just for troubleshooting? can we make this section optional with a flag if so? (always prefer to avoid creating unneeded files)

Reply:
I thought it might be useful to have if you needed it for provenance or something of the sort but agreed that it should just be optional